qt/scientific_interfaces/ISISReflectometry/GUI/Preview/PreviewModel.cpp

```diff
@@ -122,4 +137,22 @@ void PreviewModel::exportReducedWsToAds() const {
         "Could not export reduced WS. No selection has been made on the instrument viewer and/or region selector.");
   }
 }
+
+std::optional<double> PreviewModel::getThetaFromLogs(const std::string &logName) const {
+  const Mantid::API::Run &run = getLoadedWs()->run();
+  if (!run.hasProperty(logName)) {
+    return std::nullopt;
+  }
+  Property *logData = run.getLogData(logName);
+  auto logPWV = dynamic_cast<const PropertyWithValue<double> *>(logData);
+  auto logTSP = dynamic_cast<const TimeSeriesProperty<double> *>(logData);
```

Comment:
Sorry, I should have thought of this yesterday and it's just a minor thing so possibly not worth changing at this stage, but I noticed we perform this cast when we potentially don't need it. I think this section of the method could maybe be written as:

```
auto logPWV = dynamic_cast<const PropertyWithValue<double> *>(logData);
if (logPWV) {
  return *logPWV;
}

auto logTSP = dynamic_cast<const TimeSeriesProperty<double> *>(logData);
if (logTSP && logTSP->realSize() > 0) {
  return logTSP->lastValue();
}

return std::nullopt;
```

Reply:
I agree but think it's not a big enough deal to change the PR at this stage seeing as we need this merged asap